source/common/router/retry_state_impl.cc

```diff
@@ -68,6 +70,12 @@ RetryStateImpl::RetryStateImpl(const RetryPolicy& route_policy, Http::HeaderMap&
       retries_remaining_ = temp;
     }
   }
+  if (request_headers.EnvoyRetriableStatusCodes()) {
+    for (const auto code : StringUtil::splitToken(
+             request_headers.EnvoyRetriableStatusCodes()->value().getStringView(), ",")) {
+      retriable_status_codes_.emplace_back(std::stoi(std::string(code)));
```

Comment:
Do we need to check this result? I don't think this will necessarily work? This might be non-numeric and/or have white space that needs trimming? Can you add test cases that cover this?

Reply:
Afaik `std::stoi` handle most of those cases but looks like it throws an exception if it fails to convert to a number. Looks like `StringUtil::strtoul` might work better since it doesn't throw, so I'll change to that and add some tests